web/frontend/src/components/BaseMap.vue

```diff
@@ -307,12 +313,14 @@ export default defineComponent({
     map?.on('touchend', () => clearAllTimeouts());
     map?.on('move', () => clearAllTimeouts());
     setTimeout(async () => {
-      const result = await navigator.permissions.query({ name: 'geolocation' });
-      if (result.state === 'granted') {
+      const permissionState = await geolocationPermissionState();
+      if (permissionState === 'granted') {
         map?.on('load', () => {
           geolocate.trigger();
           geolocate.on('geolocate', () => {
             if (!this.$data.hasGeolocated) {
+              // prevent the default "zoom" that occurs when we automatically `trigger`
+              // the geolocate button.
```

Comment:
It took me a while to figure out what this code was about, so I thought I'd leave a comment for the next person. Is it correct?

Reply:
Yes, you got it. Thank you for documenting this because it's not obvious at ALL. I actually feel terrible that I didn't comment this because it probably took a while for you to figure out why I was doing this. :/